### PR TITLE
Fix unprivileged post start script configmap bug

### LINKED ISSUE #1912: [BUG] Webhook should have separate scirpt configmap for privilged and unprivileged mode
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**
Webhook should have separate scirpt configmap for privilged and unprivileged mode

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
Fix unprivileged post start script configmap bug

### Ⅱ. Does this pull request fix one issue?

fixes #1912 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/base/runtime_helper.go [hunk 1]
```diff
@@ -196,3 +190,31 @@ func (info *RuntimeInfo) transformTemplateWithCacheDirDisabled(template *common.
...
+	if cm == nil {
+		// ConfigMap not found, so create it
+		//todo log not found configMap
+		err = info.client.Create(context.TODO(), cmToApply)
+		if err != nil {
+			return nil
+		}
+	} else {
+		// ConfigMap found, check if updated
+		if !reflect.DeepEqual(cm.Data, cmToApply.Data) {
```
Comment (cheyang) [inline pkg/ddc/base/runtime_helper.go:209]: I have concern that if update cm in concurrent way, and some of them are failed. How to handle it?

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/ddc/base/runtime_helper.go (hunk 1 above)
Comment (TrafalgarZZZ) [reply pkg/ddc/base/runtime_helper.go]: Updated

File: pkg/ddc/base/runtime_helper.go [hunk 2]
```diff
@@ -125,7 +125,10 @@ func (info *RuntimeInfo) GetTemplateToInjectForFuse(pvcName string, option commo
 	if !found {
 		err = info.client.Create(context.TODO(), cm)
 		if err != nil {
-			return template, err
+			// If ConfigMap creation succeeds concurrently, continue to mutate
+			if otherErr := utils.IgnoreAlreadyExists(err); otherErr != nil {
```
Comment (cheyang) [inline pkg/ddc/base/runtime_helper.go:129]: Why not using `IsAlreadyExists` directly?

File: pkg/scripts/poststart/check_fuse.go [hunk 3]
```diff
@@ -61,9 +62,9 @@ echo "succeed in checking mount point $ConditionPathIsMountPoint"
 	contentUnprivilegedSidecar = `#!/bin/bash
 set -ex
 
-echo "Sending deivce ioctl to /dev/fuse"
-chmod u+x /tools/ioctl_sync
+echo "Sending device ioctl to /dev/fuse"
 /tools/ioctl_sync
```
Comment (cheyang) [inline pkg/scripts/poststart/check_fuse.go:66]: How about using `timeout /tools/ioctl_sync`, and determine the return value to check if it's timeout.

**Review by cheyang — APPROVED** [review]

**Review by cheyang — APPROVED** [review]

**Pull request merged**